src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -2376,6 +2384,22 @@ void MacroAssembler::jump_cc(Condition cc, AddressLiteral dst, Register rscratch
   }
 }
 
+void MacroAssembler::cmp32_mxcsr_std(Address mxcsr_save, Register tmp, Register rscratch) {
+  ExternalAddress mxcsr_std(StubRoutines::x86::addr_mxcsr_std());
+  assert(rscratch != noreg || always_reachable(mxcsr_std), "missing");
+
+  stmxcsr(mxcsr_save);
+  movl(tmp, mxcsr_save);
+  // Mask out any pending exceptions (only check control and mask bits)
+  if (EnableX86ECoreOpts) {
+    // On Ecore, status bits are set by default (for performance)
+    orl(tmp, 0x003f);  // On Ecore, exception bits are set by default
```

Comment:
Duplication in comment.  Comment could be modified to reflect something like "The mxcsr_std has status bits set for performance on ECore"